packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-custom-pwd-recovery-mfa/main/java/sdk.md

```diff
@@ -0,0 +1,111 @@
+> **Note**: The examples in this guide use Java 11 and Spring Boot MVC
+
+### 1: Start password recovery
+
+After the user starts the password recovery flow and selects the email authenticator for the process, Okta sends them an email matching the **Forgot Password** template you altered earlier.
+
+<div class="common-image-format">
+
+![Example of email sent to user](/img/advanced-use-cases/custom-pwd-recovery-custom-email.png "Password recovery email")
+
+</div>
+
+When the user clicks the **Reset Password** link, their browser sends a request to the endpoint defined by the template, attaching the `${oneTimePassword}` and `${request.relayState}` VTL variables as query parameters to the URL. For instance, in the sample this request might render as `http://localhost:8080/magic-link/callback?otp=726009&state=1b34371af02dd31d2bc4c48a3607cd32`.
+
+### 2: Handle the OTP and state parameters
+
+Create a callback handler that takes the `${oneTimePassword}` and `${request.relayState}` values from the query string, saves them into local variables (for instance, 'otp', and 'state'), and makes the following checks:
+
+1. That `otp` and `state` are not `null`.
+2. That `state` matches the state value stored in the current `ProceedContext` session variable.
```

Comment:
```suggestion
2. The `state` matches the state value stored in the current `ProceedContext` session variable.
```

Reply:
We're checking "that" so it doesn't make sense to change to "the". If we did change to "the" then "The state" doesn't make sense without explaining it (oddly)

Reply:
yep - I totally see that.